index.js

```diff
@@ -98,59 +98,190 @@ export default function ({
 				// If the root was not pre-rendered, add a placeholder index.html
 				// Route all requests for the index to the SSR function
 				writeFileSync(`${staticDir}/index.html`, '');
-				swaConfig.routes.push(
-					{
-						route: '/index.html',
-						rewrite: ssrFunctionRoute
-					},
-					{
-						route: '/',
-						rewrite: ssrFunctionRoute
-					}
-				);
 			}
 
+			const swaConfig = generateConfig(builder, customStaticWebAppConfig);
+
 			writeFileSync(`${publish}/staticwebapp.config.json`, JSON.stringify(swaConfig));
 		}
 	};
+	return adapter;
 }
 
 /**
- * @param {import('./types/swa').CustomStaticWebAppConfig} customStaticWebAppConfig
- * @param {string} appDir
+ * @param {import('@sveltejs/kit').Builder} builder A reference to the SvelteKit builder
+ * @param {import('./types/swa').StaticWebAppConfig} customStaticWebAppConfig Custom configuration
  * @returns {import('./types/swa').StaticWebAppConfig}
  */
-export function generateConfig(customStaticWebAppConfig, appDir) {
-	validateCustomConfig(customStaticWebAppConfig);
-
-	if (!customStaticWebAppConfig.routes) {
-		customStaticWebAppConfig.routes = [];
-	}
-
+export function generateConfig(builder, customStaticWebAppConfig = {}) {
+	builder.log.minor(`Generating Configuration (staticwebapp.config.json)...`);
 	/** @type {import('./types/swa').StaticWebAppConfig} */
 	const swaConfig = {
 		...customStaticWebAppConfig,
-		routes: [
-			...customStaticWebAppConfig.routes,
-			{
-				route: '*',
-				methods: ['POST', 'PUT', 'DELETE'],
-				rewrite: ssrFunctionRoute
-			},
-			{
-				route: `/${appDir}/immutable/*`,
-				headers: {
-					'cache-control': 'public, immutable, max-age=31536000'
-				}
-			}
-		],
 		navigationFallback: {
-			rewrite: ssrFunctionRoute
+			rewrite: ssrFunctionRoute,
+			...customStaticWebAppConfig.navigationFallback
```

Comment:
I put it in this order specifically so that custom configuration can override it, if the key is present. I am strongly in favor of being clear about the behavior in the documentation and letting the developer decide what's best for their use-case. It doesn't affect my current use-case and this is your project, so I will commit your suggestion if you feel strongly about it, but I would really like to have the opportunity to make improvements to the documentation until you're satisfied with the results.